crates/bevy_ui/src/focus.rs

```diff
@@ -123,10 +123,14 @@ pub fn ui_focus_system(
         .find_map(|window| window.cursor_position())
         .or_else(|| touches_input.first_pressed_position());
 
-    let mut moused_over_z_sorted_nodes = node_query
-        .iter_mut()
-        .filter_map(
-            |(entity, node, global_transform, interaction, focus_policy, clip, visibility)| {
+    let moused_over_nodes = ui_stack
+        .uinodes
+        .iter()
+        .rev()
```

Comment:
Why do we need to reverse here? If we need the list in the opposite order, we should be storing it sorted in the correct order (or using a double-ended queue), rather than constantly flipping it each frame.

Reply:
nit: Agreed on the sentiment, but we're not "flipping it" we're just iterating in reverse order, which I'd assume is only detrimental on very hot loops for memory access locality reasons. However I've asked myself the same question, why are we not storing it in the reverse order if that's the only way we iterate ever (but is it?).

Reply:
The UI focus system needs to traverse the UI nodes from closest to farthest, while the renderer needs to do it from farthest to closest. I'm no expert, but I believe `.rev()` on this iterator is just a reverse loop and it doesn't actually move memory around.

Reply:
Yep, just verified. Vec iteration won't force reallocation here; VecDeque is needed for double-ended insertion only.

This seems fine to me now; feel free to either leave a comment or just resolve this conversation.